src/pretalx/htmlexport/views.py

```diff
@@ -0,0 +1,54 @@
+from bakery.views import BuildableDetailView
+
+from pretalx.agenda.views.schedule import ScheduleView
+from pretalx.agenda.views.speaker import SpeakerView
+from pretalx.agenda.views.talk import TalkView
+from pretalx.person.models import SpeakerProfile
+from pretalx.schedule.models import Schedule
+from pretalx.submission.models import Submission
+
+
+class PretalxExportContextMixin():
+    def __init__(self, *args, **kwargs):
+        self._exporting_event = kwargs.pop('_exporting_event')
+        super().__init__(*args, **kwargs)
+
+    def create_request(self, *args, **kwargs):
+        request = super().create_request(*args, **kwargs)
+        request.event = self._exporting_event
+        return request
+
+    def get_context_data(self, *args, **kwargs):
+        self.object = self.get_object()  # ScheduleView crashes without this
+
+        ctx = super().get_context_data(*args, **kwargs)
+        ctx['is_html_export'] = True
+        return ctx
+
+    def get_url(self, obj):
+        return obj.urls.public
+
+    def get_queryset(self):
+        qs = super().get_queryset()
+        return qs.filter(event=self._exporting_event)
```

Comment:
Is this strictly necessary or just a guard in case we forget a filter somewhere?

Reply:
This is the main and only filter. Bakery renders all forms in the given `queryset` (e.g. "all released schedules of any event"), so this restricts the selection to the chosen event.